map/bookmark_manager.cpp

```diff
@@ -1148,10 +1148,16 @@ void BookmarkManager::SaveTrackRecording(std::string trackName)
   auto const & tracker = GpsTracker::Instance();
   CHECK(!tracker.IsEmpty(), ("Track recording should be not be empty"));
 
+  auto const trackSize = tracker.GetTrackSize();
   kml::MultiGeometry::LineT line;
-  tracker.ForEachTrackPoint([&line](location::GpsInfo const & pt, size_t id)->bool
+  kml::MultiGeometry::TimeT timestamps;
```

Comment:
There is no need to introduce temporary variables here (and do unnecessary move). Members of `kml::MultiGeometry geometry;` below can be used directly.

Reply:
As long as there are no early `throw`, `return` statements, I agree.

Reply:
Fixed!